src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1961,6 +1965,12 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                 break;
             }
 
+            if (simdSize == 64 && !IsBaselineVector512IsaSupported())
+            {
+                // We use AVX512F for all compatible base types (float, int, uint, double)
+                break;
+            }
```

Comment:
We should be able to assert this one. Unlike Vector256 which is available when only Avx is available, and so we need to query, we should have exited in the name lookup for Vector512.